src/docs/arch-rts.dox

```diff
@@ -1,113 +1,114 @@
 
-Rollback to stable is an operation that retains only the modifications that
-are stable according to the stable timestamp and recovered checkpoint snapshot.
-The checkpoint transaction snapshot details are saved at the end of every
-checkpoint are recovered and used as a recovered checkpoint snapshot.
+RTS is an operation that retains only the modifications that are considered \a stable. A
+modification is considered \a unstable if it has a durable timestamp greater than the stable
+timestamp or its transaction id is not committed according to the recovery checkpoint snapshot.
+The recovery checkpoint snapshot corresponds to the checkpoint transaction snapshot details saved at
+the end of every checkpoint.
 
-@section rts-overview Overview of rollback to stable
+RTS works for all storage types except for LSM trees.
 
-Rollback to stable scans each and every table present in the database except
-@ref arch-metadata "Metadata" to remove the modifications from the table
-that are more than stable timestamp and recovered checkpoint snapshot.
+@section rts-overview Overview of RTS
 
-In the process of removing newer modifications from the table, all the in-memory
-updates are aborted and the on-disk version updates are replaced with an update
-from history store otherwise the data store version is removed.
+RTS scans tables in the database to remove any unstable modifications. A table is selected by RTS if
+it meets one of the following conditions:
 
-Rollback to stable is performed in three phases
-1. WiredTiger startup
-2. WiredTiger shutdown
-3. Application initiated
+1. It is modified.
+2. It has prepared updates.
+3. It has updates from transactions greater than the recovery checkpoint snapshot (this is
+applicable only during the restart phase)
+4. One of the checkpoint durable timestamps (start/stop) is greater than the stable timestamp.
 
-To improve the performance of rollback to stable operation, rollback to stable
-will perform only on particular tables that need rollback. Rollback to stable
-doesn't operate on logged tables as the updates on these are stable when the
-transaction gets committed.
-
-@section rts-stable-update Stable update of rollback to stable
-
-According to rollback to stable, the stable version of update is an update that
-has durable timestamp of less than or equal to the stable timestamp and it's
-transaction id must be committed according to the checkpoint snapshot.
+On the other hand, a table is skipped by RTS if it does not meet any of the previous criteria but
+does meet one of the following:
 
-@section rts-preconditions Pre-conditions required for rollback to stable
+1. It is empty
+2. It has timestamped updates but there is no stable timestamp set.
+3. The associated files are missing.
+4. The associated files are corrupted.
+5. It is dedicated to @ref arch-metadata "metadata"
+6. It is a logged table. RTS is not necessary for those as their updates are stable when the
```

Comment:
I think we skip logged table because it is commit level durable.

Reply:
I also agree to mention `commit level` durability.

Reply:
Added in 40c7c8dd4fe0bb492d19abc02f4494c0efa285f3